pkg/pool-weighted/contracts/WeightedPoolProtocolFees.sol

```diff
@@ -65,14 +65,13 @@ abstract contract WeightedPoolProtocolFees is BaseWeightedPool, ProtocolFeeCache
     }
 
     function _getJoinExitProtocolFees(
-        bool isJoin,
         uint256[] memory preBalances,
         uint256[] memory balanceDeltas,
         uint256[] memory normalizedWeights,
         uint256 preJoinExitSupply,
         uint256 postJoinExitSupply
     ) internal view returns (uint256, uint256) {
-        uint256 preJoinExitInvariant = WeightedMath._calculateInvariant(normalizedWeights, preBalances);
+        bool isJoin = postJoinExitSupply >= preJoinExitSupply;
```

Comment:
Since this is only used once, maybe just put `postJoinExitSupply >= preJoinExitSupply` down where it's used? (Maybe with a comment, so it's still clear we're checking for join vs. exit.)

Reply:
We'll be making the comparison once for every token in the pool in that case.